packages/react/src/components/Rating/Rating.base.tsx

```diff
@@ -10,7 +10,10 @@ const getClassNames = classNamesFunction<IRatingStyleProps, IRatingStyles>();
 const RatingStar = (props: IRatingStarProps) => {
   return (
     <div className={props.classNames.ratingStar}>
-      <Icon className={props.classNames.ratingStarBack} iconName={props.icon} />
+      <Icon
+        className={props.classNames.ratingStarBack}
+        iconName={props.fillPercentage === 50 ? props.unselectedIcon || 'FavoriteStar' : props.icon}
```

Comment:
I don't think you need to provide a default for `unselectedIcon` here, since that was already handled by `Rating` itself. Also it's safer to check for a fill percentage 0 or 100, in case of ratings that aren't in 0.5 increments (like 4.3 stars) or JS math issues.
```suggestion
        // If this is a partially filled star, use the unselected icon for better contrast
        iconName={props.fillPercentage === 0 || props.fillPercentage === 100 ? props.icon : props.unselectedIcon}
```